io-amqp1_0-impl/src/main/java/org/apache/pulsar/ecosystem/io/amqp/AmqpBaseConfig.java

```diff
@@ -53,15 +56,23 @@ public static AmqpBaseConfig load(Map<String, Object> config) throws IOException
 
     public void validate() throws ConfigurationInvalidException {
 
-        Objects.requireNonNull(protocol, "The protocol property not set.");
-        Objects.requireNonNull(host, "The host property not set.");
-        if (port <= 0) {
-            throw new ConfigurationInvalidException("The configuration port is invalid.");
+        if (connection != null) {
+            if (isValidUrlComponentConfiguration()){
+                throw new ConfigurationInvalidException(
+                        "When connection is set, protocol, host, port should be empty.");
+            }
```

Comment:
Here checks all "protocol, host, port" present (and logic). Perhaps we need an or-logic?

Reply:
done